### PR TITLE
Better MultiplayerAPI error messages.

### LINKED ISSUE #22318: Add clear(er) error messages for networking
Quoting a Discord user:
*There was a problem with an rpc call I was doing. I was using rpc to call a procedure send_player_data(playerID, playerInfo), but in the rpc call, I wasn't including either of those arguments. I don't think the error message was directly caused by the bad rpc call, but it probably triggered some chain of events that eventually caused the error to rise:*
```
ERROR: Attempt to remote call/set when networking is not connected yet in SceneTree.                                       
At: scene/main/scene_tree.cpp:1760
```

### PR DESCRIPTION
**Pull request by Faless**
Possibly closes #22318 . @mhilbrunner ?

Also add a small optimization when calling a `mastersync` RPC/RSET from the master itself (will now skip the remote call, as there can only be one master).

### CODE REVIEW

File: core/io/multiplayer_api.cpp [hunk 1]
```diff
@@ -324,6 +348,7 @@ void MultiplayerAPI::_process_rset(Node *p_node, const StringName &p_name, int p
 
 void MultiplayerAPI::_process_simplify_path(int p_from, const uint8_t *p_packet, int p_packet_len) {
 
+	ERR_EXPLAIN("Invalid packet recevied. Size too small.");
```
Comment (mhilbrunner) [inline core/io/multiplayer_api.cpp]: recevied ->received, there are multiple instances of this typo, just do a replace :)

**Review by mhilbrunner — COMMENTED** [review]
Looks good besides the typo

**Comment by mhilbrunner** [comment]
And yeah this definitely is what I had in mind for #22318 :)

File: core/io/multiplayer_api.cpp (hunk 1 above)
Comment (Faless) [reply core/io/multiplayer_api.cpp]: Eheh, copy-paste is okay... unless you make a typo the first time xD

**Review by mhilbrunner — APPROVED** [review]

**Pull request merged**